utils/ratelimits.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime, timedelta
+import json
+
+from flask import abort, request
+
+
+class KeyExpired(KeyError):
+    pass
+
+
+class RatelimitCache(object):
+    def __init__(self, expire_time=timedelta(0, 1, 0)):
+        self.expire_time = expire_time
+        self.cache = {}
+
+    def __getitem__(self, item):
+        c = self.cache[item]
+
+        now = datetime.now()
+        if now - c['timestamp'] < c['expire_time']:
+            return c['data']
+
+        del self.cache[item]
+        raise KeyExpired(item)
+
+    def get(self, item):
+        c = self.cache[item]
+
+        now = datetime.now()
+        if now - c['timestamp'] < c['expire_time']:
+            return c['data']
+
+        del self.cache[item]
+        raise KeyExpired(item)
+
+    def __contains__(self, item):
+        try:
+            self[item]
```

Comment:
`return item in self.cache`. This would remove the need for try/except and fix the missing `.cache` issue?

Reply:
Resolved